src/Libraries/DSCPython/CPythonEvaluator.cs

```diff
@@ -231,6 +235,39 @@ static CPythonEvaluator()
             }
         }
 
+        /// <summary>
+        /// Processes additional bindings that are not actual inputs.
+        /// Currently, only the node name is received in this way.
+        /// </summary>
+        /// <param name="scope">Python scope where execution will occur</param>
+        /// <param name="bindingNames">List of binding names received for evaluation</param>
+        /// <param name="bindingValues">List of binding values received for evaluation</param>
+        private static void ProcessAdditionalBindings(PyScope scope, IList bindingNames, IList bindingValues)
+        {
+            const string NodeNameInput = "Name";
+            string nodeName;
+            if (bindingNames.Count == 0 || !bindingNames[0].Equals(NodeNameInput))
+            {
+                // Defensive code to fallback in case the additional binding is not there, like
+                // when the evaluator is called directly in unit tests.
+                nodeName = "USER";
+            }
+            else
+            {
+                bindingNames.RemoveAt(0);
+                nodeName = (string)bindingValues[0];
+                bindingValues.RemoveAt(0);
+            }
+
+            // Session is null when running unit tests.
+            if (ExecutionEvents.ActiveSession != null)
+            {
+                dynamic logger = ExecutionEvents.ActiveSession.GetParameterValue(ParameterKeys.Logger);
+                Action<string> logFunction = msg => logger.Log($"{nodeName}: {msg}");
+                scope.Set("DynamoPrint", logFunction.ToPython());
```

Comment:
is `ToPython()` potentially slow? Does it make sense to cache this?

Reply:
It's marshalling/encoding the function so it can be used in Python. It's the same thing we are doing for all inputs, so I wouldn't say it's particularly slow.

Reply:
okay, sounds good, I was just curious if function marshaling was slower than other types since we do this for each invocation of the python node.